tests/test_cmd_parser.py

```diff
@@ -0,0 +1,137 @@
+# -*- coding: utf-8 -*-
+from resources.lib.tubecast.youtube.utils import Command, CommandParser
+
+# Chunk containing: "c", "S", "loungeStatus", "getNowPlaying"
+cmd_bind_info = '461\n[[0,["c","cid","",8]\n]\n,[1,["S","sid"]]\n,[2,["loungeStatus",{"devices":"[{\\"app\\":\\"kodi-tubecast\\",\\"capabilities\\":\\"que,mus\\",\\"clientName\\":\\"unknown\\",\\"experiments\\":\\"\\",\\"name\\":\\"Kodi (kodi)\\",\\"id\\":\\"id\\",\\"type\\":\\"LOUNGE_SCREEN\\",\\"hasCc\\":\\"true\\"}]","connectionEventDetails":"{\\"deviceId\\":\\"id\\"}"}]]\n,[3,["getNowPlaying"]]\n]\n'
+cmd_remote_connected = '550\n[[6,["remoteConnected",{"app":"android-phone","pairingType":"dial","capabilities":"que,atp,mus","ui":"true","clientName":"android","experiments":"","name":"name","remoteControllerUrl":"","id":"id","type":"REMOTE_CONTROL","device":"{\\"app\\":\\"android-phone\\",\\"pairingType\\":\\"dial\\",\\"capabilities\\":\\"que,atp,mus\\",\\"clientName\\":\\"android\\",\\"experiments\\":\\"\\",\\"name\\":\\"name\\",\\"remoteControllerUrl\\":\\"\\",\\"id\\":\\"id\\",\\"type\\":\\"REMOTE_CONTROL\\"}"}]]\n]\n'
+cmd_remote_disconnected = '554\n[[79,["remoteDisconnected",{"app":"android-phone-14.21.54","pairingType":"dial","capabilities":"que,atp,mus","ui":"true","clientName":"android","experiments":"","name":"name","remoteControllerUrl":"","id":"id","type":"REMOTE_CONTROL","device":"{\\"app\\":\\"android-phone\\",\\"pairingType\\":\\"dial\\",\\"capabilities\\":\\"que,atp,mus\\",\\"clientName\\":\\"android\\",\\"experiments\\":\\"\\",\\"name\\":\\"name\\",\\"remoteControllerUrl\\":\\"\\",\\"id\\":\\"id\\",\\"type\\":\\"REMOTE_CONTROL\\"}"}]]\n]\n'
```

Comment:
please add the same test commands we had on test_cmds.py. I added them for a reason, they are real edge cases I received from the youtube socket

Reply:
This is rather problematic... Let me try to unwrap this:

The only difference between `cmd_1` and `cmd_2` is that the latter doesn't have the opening square brackets.
The commands are from the time the listener read the commands line by line so I think it's likely that the bracket was on the previous line:
```
[\n
20,["remoteDisconnected", ...
```

Then there's the other thing, the difference between the two commands and the one I included in the new tests. The difference is that the old commands use `\"` instead of `\\"`.
I find it really hard to explain this, so please bear with me...

Open a new python terminal and paste `cmd_1` (including the single quotes). You'll find that it turns into this:
```
'[[20,["remoteDisconnected",{"app":"android-phone-13.05.52","pairingType":"dial","capabilities":"atp,que,mus","ui":"true","clientName":"android","experiments":"","name":"MYPHONE","remoteControllerUrl":"ws://","id":"something","type":"REMOTE_CONTROL","device":"{"app":"android-phone-13.05.52","pairingType":"dial","capabilities":"atp,que,mus","clientName":"android","experiments":"","name":"MYPHONE","remoteControllerUrl":"ws://","id":"id","type":"REMOTE_CONTROL"}"}]]'
``` 
Notice how `\"` turned into `"`? That is because `' \" ' == ' " '` ([Python Reference Manual on String literals](https://docs.python.org/2.0/ref/strings.html)).
So why does the command use `\"` in the first place? Well, if you print `\\"` in Python you'll find that's exactly what happens. `\\"` turns into `\"`.
So what I think happened is that you took the command straight from the logs without accounting for the fact that you need to escape `\"` to `\\"`.

I'm sorry if this sounded condescending, I just wanted to make sure to get the point across.

If you're 100% sure that I'm wrong in both cases then we have a bit of a problem on our hands.
Take a look at the code snippet I posted above, there's no way we could parse something like `"{"app":"android-phone-13.05.52", ...}"` properly. I find it hard to believe that the YouTube app could produce such an output under any circumstances.

Reply:
Sorry for the delay, got busy with real life. I am fine with dropping those 2 tests but can you at least leave the old lines there (commented) just for reference? Thanks